README.md

```diff
@@ -1,168 +1,156 @@
+Finally, we'll note that browser-initiated prerenders fall somewhere in between these cases. In particular, the user typing `https://example.com/` in the URL bar, even before they press <kbd>Enter</kbd>, might serve as a reasonable prerender trigger, and perhaps even the prerendering could be done with credentials. The need to prevent user annoyance is still present, so the prerendering browsing context concept is important. But, what if the user types `https://example.com/logout`? Our thinking is still evolving in this area.
 
-Server behavior in response to a GET request __may not be idempotent__. Thus issuing a request early may lead to changes in server state, such as modifying data or recording an ad impression. Unauthenticated requests to servers on the public Internet are believed to be at somewhat lower risk, because the lack of credential likely does not convey significant authority to make destructive changes and because such servers already contend with web crawlers which issue such requests. In the same-origin case, an express signal from the author that the user agent should prerender is also an indication that the request is likely to be idempotent. An alternative future direction for exploration is a capability for an origin to declare that certain URLs can be accessed idempotently, for example by responding to OPTIONS requests or by serving a manifest from a well-known URL, but this suggests a higher barrier to entry.
+## Prefetching
 
-There is a risk that loading the resource may be __resource-intensive__. This is largely out of scope of this document. User agents may cancel prerendering if it has become too resource-intensive and may apply a variety of heuristics to make an educated decision about whether prerendering is likely to be lightweight.
+Although these explainers focus largely on prerendering, we expect some of the work they produce to be useful for _prefetching_ as well. TODO more.
 
-Finally, there is the risk that any proposal is __too arduous for adoption__. Unless the effect on author incentives is extremely strong it will be very difficult to convince them to deploy a complex solution. The simpler and easier to adopt, the more likely adoption will become widespread.
+## Example scenarios
 
-## Alternatives considered
+### Same-origin drop-in speedup
 
-### No declaration
+One of the simplest things a web developer can do is indicate that their site is prepared for the browser to prerender most or all of its content. They would put something like the following in their `<head>`:
 
-Why is a declaration needed at all? Why can't the document be fetched and loaded normally, just without credentials?
+```html
+<script type="speculationrules">
+{
+  "allow": [
+    {
+      "action": "prerender",
+      "url_patterns": ["/**"]
+    }
+  ],
+  "disallow": [
+    {
+      "action": "prerender",
+      "urls": ["/logout"]
```

Comment:
Possibly a signal that we need to iterate more on semantics, but the current processing model I have described in the doc this is drawn from would interpret this as prohibiting prerendering but not prefetching (allow if action <= prerender; disallow if action >= prerender, thus action==prefetch clears both).

Changing "prerender" to "prefetch" in the disallow rule fixes this.

Reply:
"action" can probably have multiple actions?  I feel that'd be less confusing, I'm a bit concerned about introducing some precedence / ordering between the modes.